index.js

```diff
@@ -262,78 +160,57 @@ Protocol.prototype.send = function send(data, cb){
     }
   }
 
-  transport.on('data', onChunk);
+  this.on('data', onChunk);
 
-  transport.write(data)
+  this._write(data)
     .catch(defer.reject);
 
   var promise = defer.promise.finally(function(){
-    transport.removeListener('data', onChunk);
+    self.removeListener('data', onChunk);
   });
 
   return nodefn.bindCallback(promise, cb);
 };
 
 Protocol.prototype.write = function(data, cb){
-  var transport = this._transport;
-
   var transmitEvents = this._transmit.parseStreamChunk(data);
   this.emit('transmit', transmitEvents);
 
   if(!this._options.echo){
     this._terminal.ignore(data);
   }
-
-  var promise = transport.write(data);
+  var promise = this._write(data);
 
   return nodefn.bindCallback(promise, cb);
 };
 
 Protocol.prototype.reset = function reset(cb){
-  var self = this;
+  var transport = this._transport;
 
-  var promise = this._setDtr()
+  var promise = transport.set({ dtr: true })
     .delay(2)
     .then(function(){
-      return self._clrDtr();
+      return transport.set({ dtr: false });
     });
 
   return nodefn.bindCallback(promise, cb);
 };
 
 Protocol.prototype.signoff = function signoff(cb){
-  var transport = this._transport;
-
   var signoffBit = new Buffer([0]);
 
-  var promise = transport.write(signoffBit);
+  var promise = this._write(signoffBit);
 
   return nodefn.bindCallback(promise, cb);
 };
 
-Protocol.prototype.open = function open(cb){
-  return this._open(cb);
-};
-
-Protocol.prototype.close = function close(cb){
-  if(this._isOpen){
-    return this._close(cb);
-  }else{
-    return nodefn.bindCallback(when.resolve(), cb);
-  }
-};
-
 Protocol.prototype.setEcho = function setEcho(echo){
   this._options.echo = echo;
   this._terminal.clearIgnore();
 };
 
 Protocol.listPorts = function(cb){
-  var results = nodefn.call(Serial.list)
-    .then(function(ports){
-      return _.pluck(ports, 'comName');
-    });
-  return nodefn.bindCallback(results, cb);
+  return nodefn.bindCallback(Transport.listPorts(), cb);
```

Comment:
this breaks the abstraction of being able to pass a transport in because it always uses the hardcoded transport